messaging/src/main/java/org/axonframework/messaging/annotation/MessageHandlingMember.java

```diff
@@ -140,9 +140,26 @@ default String signature() {
      *
      * @param annotationType Annotation to check for on the target method
      * @return {@code true} if the annotation is present on the target method, {@code false} otherwise
+     * @deprecated in favor of {@link #isA(String)}
      */
+    @Deprecated
     boolean hasAnnotation(Class<? extends Annotation> annotationType);
 
+    /**
+     * Validates whether this {@link MessageHandlingMember} is an implementation of the given {@code handlerType}.
+     * <p>
+     * Note that a given message handling member can be several handler types at once. For example an Event Handler is
+     * also a Message Handler of type Event Message.
+     *
+     * @param handlerType the {@link MessageHandlingMember} type to validate whether this handler is an implementation
+     *                    of
+     * @return {@code true} if this {@link MessageHandlingMember} is an implementation of the given {@code handlerType},
+     * {@code false} otherwise
+     */
+    default boolean isA(String handlerType) {
```

Comment:
Not sure about this method. Isn't this a question of checking whether expected attributes are present?

Reply:
That is indeed what it comes down to in the implementation. This is why the `HandlerAttributes` store everything as a `Map<String, Map<String, Object>>`. 

It is the first key in the map which contains "the type of handler" since that is the annotation class name. 
If we would combine the key of the handler and the attribute immediately, I assumed the operation would be less efficient, since we need to move through all the entries ourselves and perform a `String#contains`.

Reply:
As stated on another comment, decided against the `isA()` and `attributes()` methods, in favour of just keeping the `attribute()` method.,